### PR TITLE
tsdb/chunkenc: fix typo of return error.

### PR DESCRIPTION
**Pull request by johncming**
Fix the content of the return error. 

In the `pool.Get()` method, the error is more exact. 
Although the `FromData` is not used mostly, it's better to refactor the return error to consistent with `pool.Get`

### CODE REVIEW

**Comment by roidelapluie** [comment]
Thanks for your PR. For legal reasons, we require that all commits are signed with a [DCO](https://developercertificate.org/) before we can merge them. See [this blog post](https://www.cncf.io/blog/2016/11/08/cloud-native-software-can-trust/) for considerations around this.

This means that the last line of your commit message should read like:

`Signed-Off-By: Your Name <your@email.address>`

If you are using GitHub through the web interface, it's quickest to close this PR and open a new one with the appropriate line.

If you are using Git on the command line, it is probably quickest to amend and force push. You can do that with
````
git commit --amend --signoff
git push -f $remote $remote_branch_for_pr
````

As always, be careful when force-pushing.

**Comment by johncming** [comment]
> Thanks for your PR. For legal reasons, we require that all commits are signed with a [DCO](https://developercertificate.org/) before we can merge them. See [this blog post](https://www.cncf.io/blog/2016/11/08/cloud-native-software-can-trust/) for considerations around this.
> 
> This means that the last line of your commit message should read like:
> 
> `Signed-Off-By: Your Name <your@email.address>`
> 
> If you are using GitHub through the web interface, it's quickest to close this PR and open a new one with the appropriate line.
> 
> If you are using Git on the command line, it is probably quickest to amend and force push. You can do that with
> 
> ```
> git commit --amend --signoff
> git push -f $remote $remote_branch_for_pr
> ```
> 
> As always, be careful when force-pushing.

DCO added.

File: tsdb/chunkenc/chunk.go [hunk 1]
```diff
@@ -162,5 +161,5 @@ func FromData(e Encoding, d []byte) (Chunk, error) {
 	case EncXOR:
 		return &XORChunk{b: bstream{count: 0, stream: d}}, nil
 	}
-	return nil, fmt.Errorf("unknown chunk encoding: %d", e)
+	return nil, errors.Errorf("invalid encoding %q", e)
```
Comment (bwplotka) [inline tsdb/chunkenc/chunk.go:164]: I think it's rather nice to know that it's chunk encoding being invalid. Do you think it's not an important detail? (:

Comment (johncming) [reply tsdb/chunkenc/chunk.go:164]: It's a good question. I think that error should contain important detail.

**Comment by johncming** [comment]
I just found out in [tsdb-CHANGLOG](https://raw.githubusercontent.com/prometheus/prometheus/master/tsdb/CHANGELOG.md):

```
## 0.7.0

 - [CHANGE] tsdb now requires golang 1.12 or higher.
 - [REMOVED] `chunks.NewReader` is removed as it wasn't used anywhere.
 - [REMOVED] `FromData` is considered unused so was removed.
```

File: tsdb/chunkenc/chunk.go (hunk 1 above)
Comment (bwplotka) [reply tsdb/chunkenc/chunk.go:164]: sure, but let's add `chunk` detail?>

**Comment by bwplotka** [comment]
> [REMOVED] `FromData` is considered unused so was removed.

yea but it was added after that as some project depends on it (Thanos, Cortex) (:

File: tsdb/chunkenc/chunk.go (hunk 1 above)
Comment (johncming) [reply tsdb/chunkenc/chunk.go]: updated.

**Comment by johncming** [comment]
> > [REMOVED] `FromData` is considered unused so was removed.
> 
> yea but it was added after that as some project depends on it (Thanos, Cortex) (:

I found it in version `0.8.0` and make a typo fix(markdown code block).

**Comment by johncming** [comment]
> > [REMOVED] `FromData` is considered unused so was removed.
> 
> yea but it was added after that as some project depends on it (Thanos, Cortex) (:

I found it in version `0.8.0` and make a typo fix(markdown code block).

**Comment by johncming** [comment]
> We have a goroutine leak now :) does #7655 fix it?

OK.

**Review by bwplotka — APPROVED** [review]
Thanks!

**Pull request merged**